### PR TITLE
exchange/okex : implement bookticker for okex  , #388

### PR DESCRIPTION
**Pull request by tony1223**
implement okex bookticker event for #388

### CODE REVIEW

File: pkg/exchange/okex/stream.go [hunk 1]
```diff
@@ -343,8 +342,11 @@ func (s *Stream) read(ctx context.Context) {
 					s.EmitEvent(*et)
 
 				case *BookData:
-					s.EmitBookData(*et)
-
+					//there's "books" for 400 depth and books5 for 5 depth
+					if channel != "books5" {
+						s.EmitBookData(*et)
+					}
+					s.EmitBookTickerUpdate(et.BookTicker())
```
Comment (c9s) [inline pkg/exchange/okex/stream.go:349]: should EmitBookTickerUpdate only for `books5`?

File: pkg/exchange/okex/parse.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ type WebSocketEvent struct {
 	Arg     interface{} `json:"arg,omitempty"`
 }
 
-func parseEvent(v *fastjson.Value) (*WebSocketEvent, error) {
+func parseEvent(v *fastjson.Value) (*WebSocketEvent, string, error) {
```
Comment (c9s) [inline pkg/exchange/okex/parse.go:41]: why returning the event separately here? it's already included in the event object.

File: pkg/exchange/okex/parse.go [hunk 3]
```diff
@@ -306,26 +320,30 @@ func parseOrder(v *fastjson.Value) ([]okexapi.OrderDetails, error) {
...
 }
 
-func parseData(v *fastjson.Value) (interface{}, error) {
+func parseData(v *fastjson.Value) (interface{}, string, error) {
 
 	channel := string(v.GetStringBytes("arg", "channel"))
 
 	switch channel {
+	case "books5":
+		data, err := parseBookData(v)
```
Comment (c9s) [inline pkg/exchange/okex/parse.go:329]: I think you can include the channel in the BookData struct? so that we can keep this parse function simple.

**Comment by c9s** [comment]
Hi @tony1223, you made a merge point in your pull request, I think you need to add `--rebase` flag for your `git pull` command.

Please rebase the branch against the main branch again to remove the merge point commit~ thanks!

File: pkg/exchange/okex/stream.go (hunk 1 above)
Comment (tony1223) [reply pkg/exchange/okex/stream.go:349]: it should all fulfill the rule of bookticker in my though,but i could change it .

File: pkg/exchange/okex/parse.go (hunk 3 above)
Comment (tony1223) [reply pkg/exchange/okex/parse.go:329]: ok

**Comment by tony1223** [comment]
ok will do the rebase ~

File: pkg/exchange/okex/parse.go (hunk 2 above)
Comment (tony1223) [reply pkg/exchange/okex/parse.go]: i was thinking to make the interface consistency, haha.  i have implemented this in another way now.

**Review by c9s — APPROVED** [review]

**Comment by c9s** [comment]
awesome!

**Pull request merged**